api/documentation/controller.py

```diff
@@ -405,6 +405,95 @@ def addSchemes(self):
             }
         )
 
+        self.addComponent(
+            'schema', 'ProtocolDictSchema', 'array',
+            {
+                'items': {
+                    'type': 'object',
+                    'properties': {
+                        'name': {'type': 'string'},
+                        'label': {'type': 'string'},
+                        'description': {'type': 'string'},
+                        'instructions': {'type': 'string'},
+                        'sitewide': {'type': 'string'},
+                        'settings': {'type': 'string'},
+                        'child_settings': {'type': 'string'},
+                        'library_settings': {'type': 'string'},
+                        'supports_registration': {'type': 'string'},
+                        'supports_staging': {'type': 'string'},
+                    }
+                }
+            }
+        )
+
+        self.addComponent(
+            'schema', 'AuthServicesDictSchema', 'array',
+            {
+                'items': {
+                    'type': 'object',
+                    'properties': {
+                        'id': {'type': 'string'},
+                        'name': {'type': 'string'},
+                        'protocol': {'$ref': '#/components/schemas/ProtocolString'},
+                        'settings': {
+                            'type': 'object',
+                            'properties': {
+                                'key': {'type': 'string'}
+                            }
+                        },
+                        'libraries': {
+                            'type': 'object',
+                            'properties': {
+                                'short_name': {'type': 'string'},
+                                'key': {'type': 'string'}
+                            }
+                        }
+                    }
+                }
+            }
+        )
+
+        self.addComponent(
+            'schema', 'AdminAuthServicesSchema', 'object',
+            {
+                'admin_auth_services': {'$ref': '#/components/schemas/AuthServicesDictSchema'},
+                'protocols': {'$ref': '#/components/schemas/ProtocolDictSchema'}
+            }
+        )
+
+        self.addComponent(
+            'schema', 'AdminAuthPost', 'object',
+            {
+                'protocol': {'type': 'string'},
```

Comment:
I think this should be some kind of JSON object. We should capture that even if we don't fully capture everything that should be submitted here. The general shape of what the protocol should look like would be really useful!

Reply:
This is for the POST request on admin_auth_services and is grabbing a protocol name from a form to then search for the protocols objects.